src/grisp_tools_build.erl

```diff
@@ -330,8 +330,14 @@ repo_sanity_check(Repo, S0) ->
 run_git_plumbings([], S) ->
     {up_to_date, S};
 run_git_plumbings([{Cmd,Check}|Plumbings], S0) ->
-    {{ok, Output}, S1} = shell(S0, Cmd),
-    case Check(Output) of
-        true -> run_git_plumbings(Plumbings, S1);
-        false -> {invalid, S1}
+    try shell(S0, Cmd) of
+        {{ok, Output}, S1} ->
+            case Check(Output) of
+                true -> run_git_plumbings(Plumbings, S1);
+                false -> {invalid, S1}
+            end
+    catch
+        _Error ->
```

Comment:
I'm not a fan of these catchall handlers. Is there a way we can reliably detect that it is a connection error and not any other arbitrary Git error?

Reply:
We would have to parse the output of the git command.
Plus, inside rebar3_grisp we do not handle erroneous shell commands and we let rebar3 trigger an exception like 'rebar3_abort'. So currently from grisp_tools i cannot access the text from a failing git command.

Reply:
Then we need to do that, and fix this. Otherwise we mask any other errors or strange situations that can occur, which will make huge debug headaches.